dev/com.ibm.ws.microprofile.health.2.0_fat/fat/src/com/ibm/ws/microprofile/health20/fat/DelayAppStartupHealthCheckTest.java

```diff
@@ -70,10 +75,96 @@ public void deployApplicationAndStartServer() throws Exception {
 
     @After
     public void cleanUp() throws Exception {
-        server1.stopServer("CWWKE1102W", "CWWKE1105W", "CWMH0052W", "CWMH0053W");
+        server1.stopServer(EXPECTED_FAILURES);
+    }
+
+    @Test
+    public void testReadinessEndpointOnServerStart() throws Exception {
+        log("testReadinessEndpointOnServerStart", "Begin execution of testReadinessEndpointOnServerStart");
+        server1.setMarkToEndOfLog();
+        server1.stopServer(EXPECTED_FAILURES);
+
+        class StartServerOnThread extends Thread {
+            @Override
+            public void run() {
+                try {
+                    server1.startServer();
+                } catch (Exception e) {
+                    assertTrue("Failure to start server on a seperate thread.", server1.isStarted());
+                }
+            }
+        }
+
+        // Need to ensure the server is not finish starting when readiness endpoint is hit so start the server on a separate thread
+        // Note: this does not guarantee that we hit the endpoint during server startup, but it is highly likely that it will
+        StartServerOnThread startServerThread = new StartServerOnThread();
+        log("testReadinessEndpointOnServerStart", "Starting DelayedHealthCheck server on separate thread.");
+        startServerThread.start();
+
+        try {
+            HttpURLConnection conReady = null;
+            int responseCode = -1;
+            boolean connectionExceptionEncountered = false;
+            boolean first_time = true;
+            boolean app_ready = false;
+
+            // Repeatedly hit the readiness endpoint until a response of 200 is received
+            while (!app_ready) {
```

Comment:
we should also add some timeout component to this loop (perhaps 2 minutes max) to prevent from looping forever